diff --git a/src/eval.c b/src/eval.c
index 3b1172e..745f6d0 100644
--- a/src/eval.c
+++ b/src/eval.c
@@ -8393,7 +8393,9 @@ static struct fst
     {"range",		1, 3, f_range},
     {"readfile",	1, 3, f_readfile},
     {"reltime",		0, 2, f_reltime},
+#ifdef FEAT_FLOAT
     {"reltimefloat",	1, 1, f_reltimefloat},
+#endif
     {"reltimestr",	1, 1, f_reltimestr},
     {"remote_expr",	2, 3, f_remote_expr},
     {"remote_foreground", 1, 1, f_remote_foreground},
diff --git a/src/json.c b/src/json.c
index 7b85cd2..b628ba2 100644
--- a/src/json.c
+++ b/src/json.c
@@ -647,9 +647,9 @@ json_decode_item(js_read_T *reader, typval_T *res, int options)
 	default:
 	    if (VIM_ISDIGIT(*p) || *p == '-')
 	    {
+#ifdef FEAT_FLOAT
 		char_u  *sp = p;
 
-#ifdef FEAT_FLOAT
 		if (*sp == '-')
 		{
 		    ++sp;
